### PR TITLE
feat: add lint for recipe tests

### PR DESCRIPTION
**Pull request by nichmor**
Checklist
* [ ] Added a ``news`` entry
* [ ] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

### CODE REVIEW

File: conda_smithy/linter/rattler_linter.py [hunk 1]
```diff
@@ -20,3 +25,45 @@
     "about",
     "extra",
 ]
+TEST_KEYS = {"script", "python"}
+
+
+def lint_recipe_tests(
+    recipe_dir: Optional[str],
+    test_section: Dict[str, Any],
```
Comment (wolfv) [inline conda_smithy/linter/rattler_linter.py:33]: For rattler-build, this should be a list of dictionaries I believe.

**Comment by beckermr** [comment]
Besides @wolfv's comment, LGTM from me. Thank you for the very nice PR!

File: conda_smithy/linter/rattler_linter.py (hunk 1 above)
Comment (nichmor) [reply conda_smithy/linter/rattler_linter.py]: that's right!

**Review by beckermr — APPROVED** [review]
When we are done with these PRs, we'll need to merge a pr with a single news item that refers to the new linter and has all of the pr numbers listed.

**Pull request merged**